Sources/Themes.php

```diff
@@ -991,10 +991,6 @@ function PickTheme()
 		// Change a specific member's theme.
 		else
 		{
-			// The forum's default theme is always 0 and we
-			if (isset($_GET['th']) && $_GET['th'] == 0)
-					$_GET['th'] = $modSettings['theme_guests'];
-
```

Comment:
this breaks theme variants

Reply:
Can you describe in more detail, 
What are variant and how it break.

Reply:
Theme variants allow a single theme to have multiple colour schemes. See [here](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Themes/default/index.template.php#L95) in index.template.php, [here](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Load.php#L2150) in Load.php, and [here](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Themes.php#L686), [here](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Themes.php#L783), and [here](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Themes.php#L1145) in Themes.php

Reply:
When i set a theme as default which got a variant than the stuff get broken without this logic?

Reply:
I created some variants for the curve2 template and didn't see any issue,
by side that i'm not able to define a variant as default.
So from my side is ther no issue.

Reply:
how are you creating the variant? if you are creating a new theme then you are doing it incorrectly.
for SMF 2.0 this theme shows how it can be done https://custom.simplemachines.org/themes/index.php?lemma=2308

Reply:
i know,
but i'm not able to create this bug.
But this "workaround" create a bug which i can see.

-> So when no one is able to create the bug for this workaround,
than i see no reason to keep this.

Reply:
The bug can be fixed by moving this below updateMemberData

Reply:
Lets play your idea in our mind:
We got two styles: 
Curve2 with id 1
Lunar with id 2

The default(with id 0) shows at Curve2.
Let's assum curve2 got 3 variants and lunar 2.

Now we choose default (id 0) variant 3 (id 2),
the memberdata got as theme id 0 (this is good because we get now that what we selected)
Then the theme table get update ther we insert theme id 1 with variant id 2.

When now the admin change the default theme from curve2 to lunar,
the logic get broken.

Based on this is my guess:
You can't choose a variant on the default themen(because it can be changed every minute by the admin),
when someone choose the default the entry in the theme table needs to be deletet.